src/Network/Probe.php

```diff
@@ -141,12 +142,18 @@ private static function hostMeta($host)
 		// Reset the static variable
 		self::$baseurl = '';
 
-		$ssl_url = "https://".$host."/.well-known/host-meta";
-		$url = "http://".$host."/.well-known/host-meta";
+		// Handles the case when the hostname contains the scheme
+		if (!parse_url($host, PHP_URL_SCHEME)) {
```

Comment:
Why does `$host` can contain the scheme or not? Where is data coming from? Shouldn’t we fix the method calls instead of this?

Reply:
It is coming from the `lrdd` function. And yes, we could remove the protocol there. But doing so would force to both try HTTP and HTTPS in case of errors. So with not stripping the scheme we reduce the amount of needed requests.

Reply:
Fair.